### PR TITLE
Install Go to hostedtoolcache directory on Windows image

### PR DESCRIPTION
**Pull request by vsafonkin**
Internal issue: [#488](https://github.com/actions/virtual-environments-internal/issues/488)

Moved Go location from `C:` to `C:\hostedtoolcache\go` directory to work with `setup-go` action.

Example of path:
```
C:\hostedtoolcache\go\1.14.2\x64
```

Also it uses `7z` tool instead `Expand-Archive` to decrease expand archive time (about 20 min with `Expand-Archive` and 3 min with `7z`).

### CODE REVIEW

**Comment by maxim-lobanov** [comment]
> Also it uses 7z tool instead Expand-Archive to decrease expand archive time (about 20 min with Expand-Archive and 3 min with 7z).

That sounds really good. Found that it is [known PowerShell](https://github.com/PowerShell/Microsoft.PowerShell.Archive/issues/32) issue that is not fixed for a long time :(

Do we want to switch all `Expand-Archive` to `7zip` for all tools (in scope of separate PR)? If it gives at least 1 hour of image-generation improvement, it can be good improvement. We can implement small helper function as a wrapper for 7z invocation.
cc: @alepauly , @AlenaSviridenko for any thoughts

File: images/win/scripts/Installers/Install-Go.ps1 [hunk 1]
```diff
@@ -29,36 +29,37 @@ function Install-GoVersion
 
     # Extract the zip archive.  It contains a single directory named "go".
     Write-Host "Extracting Go $latestVersion..."
-    Expand-Archive -Path $goArchPath -DestinationPath "C:\" -Force
+    $toolDirectory = Join-Path $env:AGENT_TOOLSDIRECTORY "go\$latestVersion"
+    7z.exe x $goArchPath -o"$toolDirectory" -y | Out-Null
```
Comment (miketimofeev) [inline images/win/scripts/Installers/Install-Go.ps1:33]: `-o"$toolDirectory"` - space missed

**Comment by AlenaSviridenko** [comment]
@maxim-lobanov `7zip` sound good to me, 1 hour of win in image generation is a good speed improvement, it will allow us to check PRs faster, for example.

File: images/win/scripts/Installers/Install-Go.ps1 (hunk 1 above)
Comment (vsafonkin) [reply images/win/scripts/Installers/Install-Go.ps1:33]: I don't know why, but this flad uses without whitespace:
https://sevenzip.osdn.jp/chm/cmdline/switches/output_dir.htm

Comment (maxim-lobanov) [reply images/win/scripts/Installers/Install-Go.ps1:33]: Yep, missed space is expected here :)

File: images/win/scripts/Installers/Install-Go.ps1 [hunk 2]
```diff
@@ -29,36 +29,37 @@ function Install-GoVersion
 
     # Extract the zip archive.  It contains a single directory named "go".
     Write-Host "Extracting Go $latestVersion..."
-    Expand-Archive -Path $goArchPath -DestinationPath "C:\" -Force
+    $toolDirectory = Join-Path $env:AGENT_TOOLSDIRECTORY "go\$latestVersion"
```
Comment (alepauly) [inline images/win/scripts/Installers/Install-Go.ps1:32]: similar comment as in #848 - should we setup a link from the old location to the new while we communicate with users?

Comment (vsafonkin) [reply images/win/scripts/Installers/Install-Go.ps1:32]: Done!

File: images/win/scripts/Installers/Install-Go.ps1 [hunk 3]
```diff
@@ -29,36 +29,40 @@ function Install-GoVersion
...
     {
-        Remove-Item -Recurse -Force "C:\go\blog"
+        Remove-Item -Recurse -Force "$fullArchPath\blog"
     }
 
-    # Rename the extracted "go" directory to include the Go version number (to support side-by-side versions of Go).
-    $newDirName = "Go$latestVersion"
-    Rename-Item -path "C:\go" -newName $newDirName
+    # Create symlink in old location
+    New-Item -Path "C:\go$latestVersion" -ItemType SymbolicLink -Value $fullArchPath
```
Comment (alepauly) [inline images/win/scripts/Installers/Install-Go.ps1:51]: isn't this missing a `\`?

Comment (vsafonkin) [reply images/win/scripts/Installers/Install-Go.ps1:51]: @alepauly , no, in old location Go was installed in folders called `go1.14.2`, `go1.13.2` and etc. I have tested symlinks on created VM manually, it works correct.

**Pull request merged**